src/test/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheckTest.java

```diff
@@ -0,0 +1,228 @@
+    }
+
+    @Test
+    public void testCommentsMultiLine() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final String[] expected = {
+            "5: " + getCheckMessage(MSG_KEY, "aKey", "multi.line"),
+        };
+        verify(checkConfig, getPath("InputOrderedProperties5CommentsMultiLine.properties"),
+                expected);
+    }
+
+    @Test
+    public void testShouldNotProcessFilesWithWrongFileExtension() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verify(checkConfig, getPath("InputOrderedProperties.txt"), expected);
+    }
+
+    @Test
+    public void testIoException() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final OrderedPropertiesCheck check = new OrderedPropertiesCheck();
+        check.configure(checkConfig);
+        final String fileName =
+                getPath("InputOrderedPropertiesCheckNotExisting.properties");
+        final File file = new File(fileName);
+        final FileText fileText = new FileText(file, Collections.emptyList());
+        try {
+            final SortedSet<LocalizedMessage> messages =
+                    check.process(file, fileText);
+            fail("Expected an CheckstyleException to be thrown : " + messages);
+        }
+        catch (CheckstyleException anCheckstyleException) {
+            final String causeMessageStart = "Unable to open file";
+            assertTrue("Unable to open file ",
+                    anCheckstyleException.getMessage().startsWith(causeMessageStart));
+        }
+    }
+
+    @Test
+    public void testWrongKeyTypeInProperties() throws Exception {
+        final Class<?> orderedPropertiesClass = Class
+                .forName("com.puppycrawl.tools.checkstyle.checks."
+                        + "OrderedPropertiesCheck$SequencedProperties");
+        final Constructor<?> constructor = orderedPropertiesClass.getDeclaredConstructor();
+        constructor.setAccessible(true);
+        final Object orderedProperties = constructor.newInstance();
+        final Method method = orderedProperties.getClass().getDeclaredMethod("put", Object.class,
+                Object.class);
+        final Object result = method.invoke(orderedProperties, 1, "value");
+        final Map<Object, Object> table = new HashMap<>();
+        final Object expected = table.put(1, "value");
+        assertEquals("Invalid result of put method", expected, result);
+
+        final Object result2 = method.invoke(orderedProperties, 1, "value");
+        final Object expected2 = table.put(1, "value");
+        assertEquals("Value should be substituted", expected2, result2);
+    }
+
+    @Test
+    public void testLineNumbers() throws Exception {
```

Comment:
Why can't this use an input file instead of hacking the check?

Reply:
Hmmm lets see If I find a better solution. testLineNumbers() is private and my goal was to test the return of this method only. May be that I will keep it like this.

Reply:
Done.

Reply:
@thomassenger This isn't done because the check is still using some reflection and it hasn't been stated why you can't check the method through inputs instead of using reflection.

Reply:
It was difficult to remove. Now I found out that there is a useful report on PIT. Using it I was able to understand better what PIT is doing. The new test is testKeepForLoopIntact(). It works without reflection and trick PIT.

Reply:
Done.